sw/host/cryptotest/testvectors/parsers/wycheproof_ecdh_parser.py

```diff
@@ -0,0 +1,210 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import json
+import jsonschema
+import logging
+import sys
+
+from Crypto.Util.asn1 import DerSequence, DerObjectId, DerBitString
+
+# Mapping from the curve names used by Wycheproof to those used by cryptotest
+EC_NAME_MAPPING = {
+    "secp256r1": "p256",
+    "secp384r1": "p384",
+}
+
+# Object ID for algorithm `eccPublicKey` (https://oidref.com/1.2.840.10045.2.1)
+EccPublicKeyOid = "1.2.840.10045.2.1"
+
+EC_OID_MAPPING = {
+    # https://oidref.com/1.2.840.10045.3.1.7
+    "secp256r1": "1.2.840.10045.3.1.7",
+    # https://oidref.com/1.3.132.0.34
+    "secp384r1": "1.3.132.0.34",
+}
+
+EC_COORDINATE_SIZE_MAPPING = {
+    "secp256r1": 32,
+    "secp384r1": 48,
+}
+
+# A small number of tests include invalid OIDs in the DER that
+# Crypto.Util.asn1 fails to reject, so we manually exclude those test
+# cases. The test case IDs match what is in
+# sw/vendor/wycheproof/testvectors/ecdh_secp256r1_test.json at the
+# time of this writing. The IDs differ in the most up-to-date
+# wycheproof master branch.
+PUBLIC_KEY_BLACKLIST = [
+    # tcID: 426
+    "305a301406082a8648ce3d0201e006082a8648ce3d030107034200042998705a9a71c783e1cf4397dbed9375a44e4cb88053594b0ea982203b6363b063d0af4971d1c3813db3c7799f9f9324cbe1b90054c81b510ff6297160add66b",
+    # tcID: 427
+    "305a301406082a808648ce3d020106082a8648ce3d030107034200042998705a9a71c783e1cf4397dbed9375a44e4cb88053594b0ea982203b6363b063d0af4971d1c3813db3c7799f9f9324cbe1b90054c81b510ff6297160add66b",
+    # tcID: 428
+    "305a301406072a8648ce3d020106092a8648ce3d030107e0034200042998705a9a71c783e1cf4397dbed9375a44e4cb88053594b0ea982203b6363b063d0af4971d1c3813db3c7799f9f9324cbe1b90054c81b510ff6297160add66b",
+    # tcID: 429
+    "305a301406072a8648ce3d020106092a808648ce3d030107034200042998705a9a71c783e1cf4397dbed9375a44e4cb88053594b0ea982203b6363b063d0af4971d1c3813db3c7799f9f9324cbe1b90054c81b510ff6297160add66b",
+]
+
+def decode_asn_public_key(s, group_curve):
+    # If we have a blacklisted public key bitstring, exclude it.
+    if s in PUBLIC_KEY_BLACKLIST:
+        raise ValueError("Public key DER bitstring was blacklisted.")
+
+    s = bytes.fromhex(s)
+    
+    # SubjectPublicKeyInfo as specified in
+    # https://www.rfc-editor.org/rfc/rfc5280#section-4.1
+    pubKeyInfo = DerSequence()
```

Comment:
style nit: the python scripts generally follow PEP8 which uses snakecase with the underscores for variable names.

Reply:
Obsoleted